spec/immutable-data-structures.html

```diff
@@ -6,27 +6,590 @@ <h1>Immutable Data Structures</h1>
+      </emu-clause>
+      <emu-clause id="sec-tuple.prototype.lastIndexOf">
+        <h1>Tuple.prototype.lastIndexOf</h1>
+        <p>The initial value of the *Tuple.prototype.lastIndexOf* is the *"lastIndexOf"* property of the intrinsic object %Array.prototype%.</p>
+      </emu-clause>
+      <emu-clause id="sec-tuple.prototype.entries">
+        <h1>Tuple.prototype.entries ( )</h1>
+        <p>When the *entries* method is called, it returns an iterator over the entries of the tuple.</p>
+        <p>The following steps are taken:</p>
+        <emu-alg>
+          1. Let _O_ be ? ToObject(*this* value).
+          1. Return CreateArrayIterator(_O_, ~key+value~).
+        </emu-alg>
+      </emu-clause>
+      <emu-clause id="sec-tuple.prototype.every">
+        <h1>Tuple.prototype.every</h1>
+        <p>The initial value of the *Tuple.prototype.every* is the *"every"* property of the intrinsic object %Array.prototype%.</p>
+      </emu-clause>
+      <emu-clause id="sec-tuple.prototype.filter">
+        <h1>Tuple.prototype.filter ( _callbackfn_ [ , _thisArg_ ] )</h1>
+        <emu-note>
+          <p>_callbackfn_ should be a function that accepts three arguments and returns a value that is coercible to the Boolean value *true* or *false*. `filter` calls _callbackfn_ once for each element in the tuple, in ascending order, and constructs a new tuple of all the values for which _callbackfn_ returns *true*.</p>
+          <p>If a _thisArg_ parameter is provided, it will be used as the *this* value for each invocation of _callbackfn_. If it is not provided, *undefined* is used instead.</p>
+          <p>_callbackfn_ is called with three arguments: the value of the element, the index of the element, and the tuple being traversed.</p>
+        </emu-note>
+        <p>When the `filter` method is called with one or two arguments, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _T_ be ? thisTupleValue(*this* value).
+          1. Let _list_ be _T_'s [[Sequence]] value.
+          1. Let _len_ be the number of elements in _list_.
+          1. If IsCallable(_callbackfn_) is *false*, throw a *TypeError* exception.
+          1. Let _newList_ be a new empty List.
+          1. Let _k_ be 0.
+          1. Repeat, while _k_ &lt; _len_,
+            1. Let _kValue_ be _list_[_k_].
+            1. Let _selected_ be ! ToBoolean(? Call(_callbackfn_, _thisArg_, &laquo; _kValue_, _k_, _T_ &raquo;)).
+            1. If _selected_ is *true*, then
+              1. Append _kValue_ to the end of list _newList_.
+            1. Set _k_ to _k_ + 1.
+          1. Return a new Tuple value whose [[Sequence]] is _newList_.
+        </emu-alg>
+      </emu-clause>
+      <emu-clause id="sec-tuple.prototype.find">
+        <h1>Tuple.prototype.find</h1>
+        <p>The initial value of the *Tuple.prototype.find* is the *"find"* property of the intrinsic object %Array.prototype%.</p>

```

Comment:
```suggestion
        <p>The initial value of *Tuple.prototype.find* is %Array.prototype.find%.</p>
```

Reply:
this one still isn't resolved?

Reply:
Yeah somehow these committed suggestions got lost. I just fixed all of the occurrences of this again.